### DISCUSSION TITLE
How to access json parameters used in moleculer.config.js from docker compose environment config file (.env)

### DISCUSSION THREAD

**Discussion by rohitpatidar007** [moleculerjs/moleculer#1086] [Q&A] [answered]
Hello Everyone,
Trying to deploy molecular application using docker compose.
Has anyone got an idea how can I use docker compose env file to pass json objects to dockerized moleculer application. This is needed so that the parameters like transporter, cacher which also uses json configuration can be configured from docker compose environment file .
Ex:- for Transporter only url can be send easily as given by @icebob in moleculer demo project (mentioned as 'TRANSPORTER=nats://nats:4222' in docker-compose.env file). But how to pass all the transporter options using docker compose.

Do anyone has any suggestion on this i.e. pass json object parameters to docker container.

(I know this is more question is more related to docker compose but want to understand if anyone has done such configuration)

Thanks.

**Comment by AndreMaz** [comment]
To set deeply nested options please check this https://moleculer.services/docs/0.14/runner.html#Configuration-loading-logic

```
To overwrite broker’s deeply nested default options, which are not present in moleculer.config.js, via environment variables, use the MOL_ prefix and double underscore __ for nested properties in .env file. For example, to set the [cacher prefix](https://moleculer.services/docs/0.14/caching.html#Built-in-cachers) to MOL you should declare as MOL_CACHER__OPTIONS__PREFIX=MOL.
```

**Comment by rohitpatidar007** [comment]
Thanks @AndreMaz for the reply,
I tried above solution but I think there is an issue with moleculer in this scenario. I tried below parameters for transporter : -

MOL_TRANSPORTER__TYPE=AMQP
MOL_TRANSPORTER__OPTIONS__URL=amqp://xyz:xyz@localhost:5672/xyz
MOL_TRANSPORTER__OPTIONS__EVENTTIMETOLIVE=10000
MOL_TRANSPORTER__OPTIONS__HEARTBEATTIMETOLIVE=10000
MOL_TRANSPORTER__OPTIONS__PREFETCH=100

In above parameter options 'type', 'url' and 'prefetch' works perfectly (I can see this reflected on channel) but not eventTimeToLive/heartBeatTimeToLive (queues are created with default options). 
This might be due to code `.map(level => level.toLocaleLowerCase())` at line number 232 on https://github.com/moleculerjs/moleculer/blob/master/src/runner.js . (don't know if I'm looking at correct place)

So when it tries to read 'HEARTBEATTIMETOLIVE' from env parameter then above code in runner.js changes it to 'heartbeattimetolive', and this won't work. So the parameter options which have capital letters won't be taken into consideration.

Is this analysis correct?

CC : @icebob

**Reply by icebob** [reply]
Try `MOL_TRANSPORTER__OPTIONS__HEART_BEAT_TIME_TO_LIVE` and `MOL_TRANSPORTER__OPTIONS__EVENT_TIME_TO_LIVE`
_Double underscore nested object, single underscore capitalize_

**Comment by rohitpatidar007** [comment, marked as answer]
Hi @AndreMaz and @icebob  ,
Thanks a lot for your quick replies , I tried below as you suggested (with one change HEART_BEAT to HEARTBEAT) and it works ( hope I should have seen line 240 in runner.js :) prior ): 
MOL_TRANSPORTER__OPTIONS__EVENT_TIME_TO_LIVE=10000
MOL_TRANSPORTER__OPTIONS__HEARTBEAT_TIME_TO_LIVE=10000

Thanks Again Buddy.
